subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryFileTreeTest.groovy

```diff
@@ -234,23 +263,26 @@ class DirectoryFileTreeTest extends AbstractProjectBuilderSpec {
 
         private void handleDetails(FileVisitDetails details) {
             def file = details.getFile()
-            assert expectedVisitations[0].contains(file)
-            expectedVisitations[0].remove(file)
-            if (expectedVisitations[0].isEmpty() || file == stopOn) {
-                expectedVisitations.remove(0)
-                if (file == stopOn) {
-                    details.stopVisiting()
-                }
+            def found = expectedVisitations.find { it.remove(file) }
```

Comment:
Let's not modify something within the find.

Reply:
removed